novice-goals.Rmd

```diff
@@ -0,0 +1,226 @@
+    -   line plot
+    -   histogram
+    -   bar plot
+    -   which to use when
+-   How can I plot multiple datasets at once?
+    -   grouping
+    -   faceting
+-   How can I make misleading plots?
+    -   showing a single central tendency data point instead of the individual observations
+    -   saturated plots instead of for example violins or 2D histograms
+    -   not starting axis at zero to exaggerate a difference
+    -   not using perceptually uniform colormaps to indicate quantities
+    -   not thinking about color blindness
+-   What *isn't* included?
+    -   outliers
+    -   interactive plots
+    -   maps
+    -   3D visualization
+
+## Development
+
+-   How can I make my own functions?
+    -   declaring functions
+    -   declaring parameters
+    -   default values
+    -   common conventions
+-   How can I make my programs tell me that something has gone wrong?
+    -   validation (did we build the right thing) vs. verification (did we build the thing correctly)
+    -   assertions for sanity checks
+-   How can I ask for help?
+    -   creating a reproducible example (reprex)
+-   How do I install software?
+    -   what *is* a package?
+    -   package manager
+-   What *isn't* included?
+    -   code browsers, multiple cursors, and other fancy IDE tricks
+    -   virtual environments
+    -   debuggers
+
+## Data analysis
+
+-   How can I represent and manage missing values?
+    -   NA
+-   How can I get a feel for my data?
+    -   summary statistics
+-   How can I create a simple model of my data?
+    -   formulas
+    -   linear regression
+        -   adding a best fit straight line on a scatterplot
+        -   understanding what the error bands on a "best fit" straight line mean
+    -   k-means cluster analysis
+    -   frame these as exploratory tools for revealing structure in the data, rather than modelling or inferential tools
+-   How can I put people at risk?
+    -   algorithmic bias
+    -   de-anonymization
+-   What *isn't* included?
+    -   statistical tests
+    -   multiple linear regression
+    -   anything with "machine learning" in its name
+
+## Version control
+
+-   What is a version control system?
+    -   a smarter kind of backup
+-   What goes where and why?
+    -   local vs. remote storage (physically)
+    -   local vs. remote storage (ethical/privacy issues)
+-   How do I track my work locally?
+    -   diff
+    -   add
+    -   commit
+    -   log
+-   How do I view or recover an old version of a file?
+    -   diff
+    -   checkout
+-   How do I save work remotely?
+    -   push and pull
+-   How do I manage conflicts?
+    -   merge
+-   What *isn't* included?
+    -   forking
+    -   branching
+    -   pull requests
+    -   git reflow --substantive --single-afferent-cycle --ia-ia-rebase-fhtagn ...
+
+## Publishing
+
+-   How do static websites work?
+    -   URLs
+    -   servers
+    -   request/response cycle
+    -   pages
+-   How do I create a simple HTML page?
```

Comment:
How in depth are we thinking about for this? Just enough to know what is happening when Markdown is converted to HTML?

Reply:
Yup.